website/docs/test_core_lib.md

```diff
@@ -0,0 +1,172 @@
+---
+id: test_core_lib
+title: Test Core Lib
+sidebar_label: Test Core Lib
+---
+
+Let's understand how `Core-Lib` is initialized and how to integrate it with your new or existing application.
+
+## Main Class
+Here you'll have all the `DataAccess`, `Service`,  `Connection`, `Cache` initialized. Which can be further accessed when we initialize the `Core-Lib`.
+
+`user_core_lib.py`
+```python
+from omegaconf import DictConfig
+
+from core_lib.cache.cache_handler_ram import CacheHandlerRam
+from core_lib.core_lib import CoreLib
+from core_lib.connection.sql_alchemy_connection_registry import SqlAlchemyConnectionRegistry
+
+from user_core_lib.data_layers.data_access.customer_data_access import CustomerDataAccess
+from user_core_lib.data_layers.data_access.user_data_access import UserDataAccess
+from user_core_lib.data_layers.service.customer_service import CustomerService
+from user_core_lib.data_layers.service.user_service import UserService
+
+
+class UserCoreLib(CoreLib):
+    def __init__(self, conf: DictConfig):
+        super().__init__()
+        self.config = conf
+        CoreLib.cache_registry.register("memory_cache", CacheHandlerRam())
+        db_session = SqlAlchemyConnectionRegistry(self.config.core_lib.user_core_lib.data.userdb)
+        self.user = UserService(UserDataAccess(db_session))
+```
+
+## DataAccess
+Here you should define the functions that only and only communicate with the connection created, for e.g., queries to get some data from a database or insert some data in the database, Solr queries or whatever the function has to interface with the connection.
+
+`user_data_access.py`
+```python
+from http import HTTPStatus
+
+from core_lib.data_layers.data_access.data_access import DataAccess
+from core_lib.connection.sql_alchemy_connection_registry import SqlAlchemyConnectionRegistry
+from core_lib.error_handling.status_code_exception import StatusCodeException
+from user_core_lib.data_layers.data.db.user import User
+from core_lib.error_handling.not_found_decorator import NotFoundErrorHandler
+
+
+class UserDataAccess(DataAccess):
```

Comment:
can we just make this CRUD? fewer lines of code. and the data access is not the main focus here